runtime/src/locked_accounts_results.rs

```diff
@@ -17,6 +17,10 @@ impl<'a, 'b> LockedAccountsResults<'a, 'b> {
         transactions: &'b [Transaction],
         txs_iteration_order: Option<Vec<usize>>,
     ) -> Self {
+        assert_eq!(locked_accounts_results.len(), transactions.len());
```

Comment:
I added a couple asserts here to capture the implicit assumption that all three lists are of equal length